src/scripts/main.js

```diff
@@ -1,3 +1,183 @@
 'use strict';
 
-// write your code here
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLoose = document.querySelector('.message-lose');
+
+const button = document.querySelector('.button');
+
+const gameScore = document.querySelector('.game-score');
+
+const boardRows = document.querySelectorAll('.field-row');
+
+let score = 0;
+let gameBoard = boardInit();
+
+function boardInit() {
+  return [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+}
+
+function resetGame() {
+  gameBoard = boardInit();
+  score = 0;
+  messageLoose.classList = 'message message-lose hidden';
+  messageWin.classList = 'message message-win hidden';
+}
+
+function renderGameBoard() {
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 4; col++) {
+      boardRows[row].children[col].innerText
+        = gameBoard[row][col] ? gameBoard[row][col] : '';
+
+      boardRows[row].children[col].className
+        = `field-cell field-cell--${gameBoard[row][col]}`;
+    }
+  }
+  gameScore.innerText = score;
+}
+
+function setRandomCell() {
+  const [x, y] = [
+    getRandomNumber(),
+    getRandomNumber(),
+  ];
+
+  if (!gameBoard[x][y]) {
+    gameBoard[x][y] = Math.random() >= 0.9 ? 4 : 2;
+    gameCheck();
+  } else {
+    setRandomCell();
+  }
+}
+
+function getRandomNumber() {
+  return Math.floor(Math.random() * 4);
+}
```

Comment:
```suggestion
function getRandomNumber() {
  return [
    Math.floor(Math.random() * 4),
    Math.floor(Math.random() * 4)
  ]
}
```
Можна було б повертати tuple з цієх функції і вище писати один раз
`const [x, y] = getRandomNumber()`